kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DeltaTableReadsSuite.scala

```diff
@@ -38,6 +39,20 @@ import scala.collection.JavaConverters._
 
 class DeltaTableReadsSuite extends AnyFunSuite with TestUtils {
 
+  test("CRC metadata & protocol fetching") {
+    withTempDir { dir =>
+      val path = dir.getCanonicalPath + "/table"
+      spark.conf.set(DeltaSQLConf.DELTA_WRITE_CHECKSUM_ENABLED.key, true)
```

Comment:
you can use `withSQLConf` for tests instead of setting it (this takes care of unsetting any confs after)